crates/testing/src/consistency_task.rs

```diff
@@ -253,6 +282,92 @@ impl<TYPES: NodeType<BlockHeader = TestBlockHeader>, V: Versions> ConsistencyTas
 
         Ok(())
     }
+
+    async fn partial_validate(&self) -> Result<TestProgress> {
+        self.check_view_success().await?;
+        self.check_view_failure().await?;
+
+        self.check_total_successes().await
+    }
+
+    fn add_error(&mut self, error: Error) {
+        self.errors.push(error);
+    }
+
+    async fn handle_result(&mut self, result: Result<TestProgress>) {
+        match result {
+            Ok(TestProgress::Finished) => {
+                let _ = self.test_sender.broadcast(TestEvent::Shutdown).await;
+            }
+            Err(e) => {
+                self.add_error(e);
+                let _ = self.test_sender.broadcast(TestEvent::Shutdown).await;
+            }
+            Ok(TestProgress::Incomplete) => {}
+        }
+    }
+
+    async fn check_total_successes(&self) -> Result<TestProgress> {
+        let sanitized_network_map = sanitize_network_map(&self.consensus_leaves)?;
+
+        let inverted_map = invert_network_map::<TYPES, V>(&sanitized_network_map).await?;
+
+        if inverted_map.len() >= self.safety_properties.num_successful_views {
+            Ok(TestProgress::Finished)
+        } else {
+            Ok(TestProgress::Incomplete)
+        }
+    }
+    pub async fn check_view_success(&self) -> Result<()> {
```

Comment:
This checking that we didn't decide a view that we think should fail? can't we just check this super simply by looking at the `leaf.view` rather than getting it back out of our map